### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by nilab-khwaja**
- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsibility Principle".
- Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -11,6 +11,36 @@ def initialize
...
+    when 2
+      list_all_people
+    when 3
+      create_person
+    when 4
+      add_book
+    when 5
+      add_rental
+    when 6
+      list_all_rentals
```
Comment (mckent05) [inline app.rb:39]: - Good job done so far, but I have an error each time I try to list all rentals for a particular ID. Please fix this and ensure there are no errors. A reason for this error is that when creating a rental in your `add_rental` method you have passed in your attributes in the wrong order. Your rental class takes in the `book`, `person` and `date` attributes in that order. 

Your project:
rergawerg

File: app.rb [hunk 2]
```diff
@@ -11,6 +11,36 @@ def initialize
...
+    else
+      'Wrong number please enter a number between 1 and 7'
+    end
+  end
+
```
Comment (mckent05) [inline app.rb]: - You guys have done well so far, but please note that you have not properly refactored all of your code, as you'd notice the app.rb file is still bulky and does not follow the [single-responsibility-principle](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid), which helps us ensure that our class or module should be responsible for a single action. According to the project requirements, I suggest that you extract some of your logic in the app.rb into separate classes and ensure that each of your new classes follows the SRP. Please fix this 🙏

**Review by mckent05 — CHANGES_REQUESTED** [review]
Hi @nilab-khwaja, 

# STATUS: CHANGES REQUIRED ♻️ ♻️ 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
almost there

## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👍 👍 
- Good job refactoring your project 👍 👍 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by IngDavidTM — CHANGES_REQUESTED** [review]
## Hi Team 🙋‍♂️, 

Good job so far!
You have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!



## Top highlight 🎯
- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 

## Suggested changes

- [ ] Although you performed admirably, there are certain issues concerning the listing of rentals for a given ID. Whenever I attempt to view the rentals added for a specific user, it consistently shows a message stating "No rentals found for the specified person." Therefore, I kindly request that you address and rectify this issue.

Screenshot from 2023-11-13 12-38-24


- [x] _Check the comments under the review._ ⬇

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by nilab-khwaja** [comment]
@IngDavidTM  HI,   Thanks for your review could you please give some hint to how to solve that , we tried a lot but wasn't be able to solve that . Thanks

**Comment by mckent05** [comment]
> @IngDavidTM HI, Thanks for your review could you please give some hint to how to solve that , we tried a lot but wasn't be able to solve that . Thanks

Hello @nilab-khwaja , @Izebiafe , I forgot to mention this issue during my review, if you have not fixed the issue yet, here's a hint on how to fix it. In your  `list_all_rentals` method in app.rb, please use the `.to_i` method instead of the `.to_s` method when getting the user input for the `id`. I hope this solves your issue 😄

**Review by Olamarx — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Linters passe ✔️
- Changes requested fixed ✔️
- Readme file professional ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**